Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -1547,11 +1631,20 @@ protected function showTermsOfService()
                         ilUtil::redirect('index.php?target=' . $_GET['target'] . '&client_id=' . CLIENT_ID);
                     }
                 }
-
+                global $DIC;
```

Comment:
Please don't use the `global` keyword here (it is not a static method, so this should be doable). Instead, use the `$DIC` only in `__construct`, fetch the required services from the container, and assign them to member variables.